File: python/sort/leetcode/merge_intervals.py
from typing import List


# 合并区间
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        res = []
        for v in intervals:
            if not res or res[-1][1] < v[0]:
                res.append(v)
            else:
                res[-1][1] = max(res[-1][1], v[1])
        return res

File: go/sort/leetcode/merge_intervals.go
package leetcode

import (
	"sort"
)

//合并区间

func merge(intervals [][]int) [][]int {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})

	var res [][]int
	for i := 0; i < len(intervals); i++ {
		if len(res) == 0 || res[len(res)-1][1] < intervals[i][0] {
			res = append(res, intervals[i])
		} else if intervals[i][1] > res[len(res)-1][1] {
			res[len(res)-1][1] = intervals[i][1]
		}
	}
	return res
}
